lib/services/git-service.ts

```diff
@@ -94,43 +87,42 @@ export class GitService implements IGitService {
         });
     }
 
-    private getResourcesFromSource(templateName: string, assetDictionary: any, versionTag: string) {
-        const content: any = {};
-        const promises: Array<any> = [];
-
+    private getResourcesFromSource(templateName: string, assetDictionary: any, versionTag: string): Promise<any> {
         return new Promise((resolve, reject) => {
+            const promisesMap: Map<string, Promise<any>> = new Map();
+
             for (const key in assetDictionary) {
                 if (assetDictionary.hasOwnProperty(key)) {
-                    promises.push(
-                        util.request({
-                            method: "GET",
-                            // tslint:disable-next-line:max-line-length
-                            uri: util.format("https://raw.githubusercontent.com/NativeScript/%s/%s/tools/assets/%s", templateName, versionTag, assetDictionary[key]),
-                            resolveWithFullResponse: true,
-                            encoding: "binary",
-                            headers: util.defaultHeaders
-                        })
-                            .then((response: any) => {
-                                // tslint:disable-next-line:max-line-length
-                                content[key] = "data:image/png;base64," + new Buffer(response.body.toString(), "binary").toString("base64");
-                            })
-                            .catch((error: any) => {
-                                return {
-                                    message: "Error retrieving " + templateName + " assets from source",
-                                    err: error
-                                };
-                            }));
-
+                    promisesMap.set(key, this.getResource(templateName, versionTag, assetDictionary[key]));
                 }
             }
-            Promise.all(promises)
-                .then(() => {
-                    resolve(content);
-                })
-                .catch((error) => {
-                    reject(error);
+
+            resolve(promisesMap);
```

Comment:
This does not seem right to me -- we are resolving something but then we are continuing execution with the then(...) statement?

Also we are not handling the rejection case?